src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/model/ScanSettings.java

```diff
@@ -57,4 +58,16 @@ public SortedSet<ScanType> getScanTypes() {
     public void setScanTypes(final SortedSet<ScanType> scanTypes) {
         this.scanTypes = scanTypes;
     }
+
+    public ScanType getScanTypeWithName(String scanTypeName) {
+        Optional<ScanType> scanType = scanTypes.stream().filter(scanTool -> scanTool.getScanTypeName().equals(scanTypeName)).findFirst();
+
+        if(scanType.isPresent()) {
+            return scanType.get();
+        } else {
+            ScanType newScanType = new ScanType(scanTypeName, new TreeMap<>(), new TreeSet<>());
+            scanTypes.add(newScanType);
+            return newScanType;
+        }
+    }
```

Comment:
This function is just finding a scan Type if it exists otherwise will create a new one.